testing.py

```diff
@@ -184,11 +192,20 @@ def _work(self, title=None, authors=None, genre=None, language=None,
             genre, ignore = Genre.lookup(self._db, genre, autocreate=True)
         work.genres = [genre]
         work.random = 0.5
+
         work.editions = [primary_edition]
         primary_edition.is_primary_for_work = True
-        work.primary_edition = primary_edition
+
+        #work.primary_edition = primary_edition
+        work.set_primary_edition()
+
         if pool != None:
-            work.license_pools.append(pool)
+            # make sure the pool's presentation_edition is set, 
+            # bc loan tests assume that.
+            pool.set_presentation_edition(None)
+
+            if not work.license_pools[0]:
```

Comment:
Changed.  I had a problem where the work got two identical license pools, added at different points in the code, and this line makes sure there's no duplication.